crates/vm/levm/src/gas_cost.rs

```diff
@@ -720,3 +720,43 @@ pub fn staticcall(
         .checked_add(dynamic_gas)
         .ok_or(OutOfGasError::GasCostOverflow)?)
 }
+
+pub fn fake_exponential(factor: u64, numerator: u64, denominator: u64) -> Result<U256, VMError> {
+    let mut i = 1;
+    let mut output: u64 = 0;
+
+    // Initial multiplication: factor * denominator
+    let mut numerator_accum = factor.checked_mul(denominator).ok_or(VMError::Internal(
+        InternalError::ArithmeticOperationOverflow,
+    ))?;
+
+    while numerator_accum > 0 {
+        // Safe addition to output
+        output = output
+            .checked_add(numerator_accum)
+            .ok_or(VMError::Internal(
+                InternalError::ArithmeticOperationOverflow,
+            ))?;
+
+        // Safe multiplication and division within loop
+        numerator_accum = numerator_accum
+            .checked_mul(numerator)
+            .ok_or(VMError::Internal(
+                InternalError::ArithmeticOperationOverflow,
+            ))?
+            .checked_div(denominator.checked_mul(i).ok_or(VMError::Internal(
+                InternalError::ArithmeticOperationOverflow,
+            ))?)
+            .ok_or(VMError::Internal(
+                InternalError::ArithmeticOperationOverflow,
+            ))?;
+
+        i = i.checked_add(1).ok_or(VMError::Internal(
+            InternalError::ArithmeticOperationOverflow,
+        ))?;
+    }
+
+    Ok(U256::from(output.checked_div(denominator).ok_or(
+        VMError::Internal(InternalError::ArithmeticOperationOverflow),
+    )?))
+}
```

Comment:
Nit. Make use of the advantage of implementing `#[from]` in `thiserror` errors.
```suggestion
pub fn fake_exponential(factor: u64, numerator: u64, denominator: u64) -> Result<U256, VMError> {
    let mut i = 1;
    let mut output: u64 = 0;

    // Initial multiplication: factor * denominator
    let mut numerator_accum = factor.checked_mul(denominator).ok_or(InternalError::ArithmeticOperationOverflow)?;

    while numerator_accum > 0 {
        // Safe addition to output
        output = output
            .checked_add(numerator_accum)
            .ok_or( InternalError::ArithmeticOperationOverflow)?;

        // Safe multiplication and division within loop
        numerator_accum = numerator_accum
            .checked_mul(numerator)
            .ok_or(InternalError::ArithmeticOperationOverflow)?
            .checked_div(denominator.checked_mul(i).ok_or(InternalError::ArithmeticOperationOverflow)?)
            .ok_or(VMError::Internal(InternalError::ArithmeticOperationOverflow)?;

        i = i.checked_add(1).ok_or(InternalError::ArithmeticOperationOverflow)?;
    }

    Ok(U256::from(output.checked_div(denominator).ok_or(InternalError::ArithmeticOperationOverflow)?))
}
```

Reply:
https://github.com/lambdaclass/ethrex/pull/1345/commits/92e4848a29c731291131fc3b9b3fb353bd38b369